x/stake/types/msg.go

```diff
@@ -27,36 +27,53 @@ var maximumBondingRationalDenominator sdk.Int = sdk.NewInt(int64(math.Pow10(MaxB
 // MsgCreateValidator - struct for unbonding transactions
 type MsgCreateValidator struct {
 	Description
-	ValidatorAddr  sdk.Address   `json:"address"`
-	PubKey         crypto.PubKey `json:"pubkey"`
-	SelfDelegation sdk.Coin      `json:"self_delegation"`
+	DelegatorAddr sdk.Address   `json:"delegator_address"`
+	ValidatorAddr sdk.Address   `json:"validator_address"`
+	PubKey        crypto.PubKey `json:"pubkey"`
+	Delegation    sdk.Coin      `json:"delegation"`
 }
 
+// Default way to create validator. Delegator address and validator address are the same
 func NewMsgCreateValidator(validatorAddr sdk.Address, pubkey crypto.PubKey,
 	selfDelegation sdk.Coin, description Description) MsgCreateValidator {
 	return MsgCreateValidator{
-		Description:    description,
-		ValidatorAddr:  validatorAddr,
-		PubKey:         pubkey,
-		SelfDelegation: selfDelegation,
+		Description:   description,
+		DelegatorAddr: validatorAddr,
+		ValidatorAddr: validatorAddr,
+		PubKey:        pubkey,
+		Delegation:    selfDelegation,
+	}
+}
+
+// Creates validator msg by delegator address on behalf of validator address
+func NewMsgCreateValidatorOnBehalfOf(delegatorAddr, validatorAddr sdk.Address, pubkey crypto.PubKey,
+	delegation sdk.Coin, description Description) MsgCreateValidator {
+	return MsgCreateValidator{
+		Description:   description,
+		DelegatorAddr: delegatorAddr,
+		ValidatorAddr: validatorAddr,
+		PubKey:        pubkey,
+		Delegation:    delegation,
 	}
 }
 
 //nolint
 func (msg MsgCreateValidator) Type() string { return MsgType }
 func (msg MsgCreateValidator) GetSigners() []sdk.Address {
-	return []sdk.Address{msg.ValidatorAddr}
+	return []sdk.Address{msg.DelegatorAddr}
 }
 
 // get the bytes for the message signer to sign on
 func (msg MsgCreateValidator) GetSignBytes() []byte {
 	b, err := MsgCdc.MarshalJSON(struct {
 		Description
-		ValidatorAddr string   `json:"address"`
+		DelegatorAddr string   `json:"delegator_address"`
+		ValidatorAddr string   `json:"validator_address"`
 		PubKey        string   `json:"pubkey"`
 		Bond          sdk.Coin `json:"bond"`
```

Comment:
Can we fix this in this PR? just notices bond should probably be `Delegation` and it's value should probably be included in the field input value just below?

Reply:
Yup, also the term Bond is used in MsgDelegate as well. Should that be changed too Delegation too?

Reply:
yes